src/System.Private.CoreLib/shared/System/Buffer.cs

```diff
@@ -372,7 +391,7 @@ private static void Memmove(ref byte dest, ref byte src, nuint len)
             if (len <= 32)
                 goto MCPY01;
 #if HAS_CUSTOM_BLOCKS
-            Unsafe.As<byte, Block16>(ref Unsafe.Add(ref dest, 16)) = Unsafe.As<byte, Block16>(ref Unsafe.Add(ref src, 16)); // [0,32]
+            Unsafe.As<byte, Vector128<byte>>(ref Unsafe.Add(ref dest, 16)) = Unsafe.As<byte, Vector128<byte>>(ref Unsafe.Add(ref src, 16)); // [0,32]
```

Comment:
Given that we pin for long arrays, and have handling for short arrays, is there any reason we can't just pin for the medium array length case as well?

`Unsafe.As<byte, Vector128<byte>>(ref Unsafe.Add(ref src, 16));` is hard to read and easy to get wrong.

Reply:
There are two different methods one that takes pointers; and this one that takes `ref` and doesn't pin anything

Reply:
It pins if it hits the native fallback case (long arrays), where it calls `_Memmove`.